timestamp-api/src/main/java/com/palantir/timestamp/TimestampStoreInvalidator.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.timestamp;
+
+public interface TimestampStoreInvalidator {
+    /**
+     * This method performs a backup of the data in an underlying timestamp bound store, and then corrupts it
+     * (if this has not already been performed). It then returns the value that has been backed up.
+     * These operations are grouped because of the requirement that as far as the timestamp bound store is concerned,
+     * this operation should be atomic. Otherwise we can have the following situation:
+     *
+     * 1. Client A reads a backup timestamp B1, preparing to invalidate and backup B1
+     * 2. Client B issues the timestamp B1 + 1
+     * 3. Client A invalidates the internal store and fast forwards a different timestamp service to B1
```

Comment:
Yep. I will add a few tests to C*TSInvalidatorIntegrationTest that simulate this (i.e. multiple invalidators pointing at the same KVS).